File: Leetcode/1769. Minimum Number of Operations to Move All Balls to Each Box.py
class Solution:
    def minOperations(self, boxes: str) -> List[int]:
        n=len(boxes)
        l=[0]*n
        left=1 if boxes[0]=='1' else 0
        right=0
        for i in range(1,n):
            if boxes[i]=='1':
                l[0]+=i
                right+=1
        for i in range(1,n):
            l[i]=l[i-1]+left-right
            if boxes[i]=='1':
                left+=1
                right-=1
        return l

File: Leetcode/1769. Minimum Number of Operations to Move All Balls to Each Box.cpp
class Solution {
public:
    vector<int> minOperations(string boxes) {
        int n=boxes.size();
        vector<int> res(n,0);
        int r=0,l=(boxes[0]=='1')?1:0;
        for(int i=1;i<n;i++){
            if(boxes[i]=='1'){
                res[0]+=i;
                r++;
            }
        }
        for(int i=1;i<n;i++){
            res[i]=res[i-1]+l-r;
            if(boxes[i]=='1'){
                l++;
                r--;
            }
        }
        return res;
    }
};
